### DISCUSSION TITLE
Integration testing with NextJS 12+ Middleware

### DISCUSSION THREAD

**Discussion by kristojorg** [vercel/next.js#41262] [Help]
I have an app that performs authentication in NextJS 12 Middleware which I would like to integration test using Cypress. I can successfully intercept requests to external APIs using [mock-service-worker](https://mswjs.io/) on both the client side and server side. However, I'm not able to intercept fetch requests made by my `middleware.ts` file. This means that I cannot mock out authentication, and my tests have to actually sign a real user in and depend on third party functionality.

I am wondering if anyone has had success intercepting requests from NextJS middleware? I have been trying to do this by running my NextJS app programmatically when starting Cypress, and starting a MSW interception service at the same time in both the client and the server. I can intercept requests from `getStaticProps`, but not from `middleware.ts`. Any ideas?

**Comment by kristojorg** [comment]
I have opened a ticket with MSW https://github.com/mswjs/msw/discussions/1429 to support the edge runtime. They have said this is not on their roadmap currently as there is no standard edge runtime.

**Comment by hamzaboularbah** [comment]
Is there any news regarding this? i am running in the same issue :/

**Comment by yashboura303** [comment]
Hi I know it's late, but do you have an example of how you achieved interception on server side call (getInitialProps)? I am still here struggling to do that, [this](https://github.com/vercel/next.js/tree/canary/examples/with-msw) doesn't work for me
